src/test/java/org/jenkinsci/plugins/workflow/cps/steps/RestartingLoadStepTest.java

```diff
@@ -157,4 +159,55 @@ public void pauseInsideLoad() throws Exception {
         });
     }
 
+    @Issue("JENKINS-50172")
+    @Test public void loadAndUnnamedClassesInPackage() {
```

Comment:
This testcase is fine for just *package* collisions but should be extended to cover classname collisions too.

Reply:
Yeah, like in the JIRA, that isn’t actually a thing. =)